src/snowflake/snowpark/stored_procedure.py

```diff
@@ -790,7 +793,7 @@ def _do_register_sp(
                     object_name=udf_name,
                     all_imports=all_imports,
                     all_packages=all_packages,
-                    is_temporary=stage_location is None,
+                    is_temporary=not is_permanent,
```

Comment:
I wonder if it makes sense to rename `is_temporary` to `is_permanent`? This way we only need to do negation once instead of in all four classes.

We probably should have made the public param `is_temporary` at the beginning so we don't need this awkward flip lol. But now all is done 🤷

Reply:
we could rename `is_temporary -> is_permanent`. public API already has `is_permanent` so we shouldn't change that.